app/js/Router.js

```diff
@@ -19,76 +8,94 @@ const SmokeTest = require('smoke-test-sharelatex')
 const AuthController = require('./AuthController')
 const HttpController = require('./HttpController')
 const SessionMiddleware = require('./SessionMiddleware')
+const { rateLimit } = require('./RateLimitMiddleware')
 
-module.exports = Router = {
-  initialize(app) {
-    app.use(express.static('public'))
-    app.use(SessionMiddleware.middleware)
-    app.use(BodyParser.urlencoded({ extended: false }))
+module.exports = {
+  initialize
+}
+
+function initialize(app) {
+  app.use(express.static('public'))
+  app.use(SessionMiddleware.middleware)
+  app.use(BodyParser.urlencoded({ extended: false }))
 
-    app.get('/', HttpController.home)
+  app.get('/', HttpController.home)
 
-    app.post(
-      '/login',
-      celebrate({
-        body: Joi.object({
-          email: Joi.string()
-            .trim()
-            .email()
-            .required(),
-          password: Joi.string()
-            .trim()
-            .required()
-        })
-      }),
-      AuthController.login,
-      AuthController.handleLoginErrors
-    )
+  app.post(
+    '/login',
+    celebrate({
+      body: Joi.object({
+        email: Joi.string()
+          .trim()
+          .email()
+          .required(),
+        password: Joi.string()
+          .trim()
+          .required()
+      })
+    }),
+    rateLimit({
```

Comment:
Should we set `skipSuccessfulRequests:true` on successful login - presumably we only need to rate limit failed requests? [express-rate-limit#skipsuccessfulrequests](https://github.com/nfriedly/express-rate-limit#skipsuccessfulrequests).  This may help to reduce support issues with institutional proxies.